test/helpers/wait-for.js

```diff
@@ -25,7 +25,7 @@ function waitForFunction(fn, arg, options = {}) {
 
       try {
         result = fn(arg);
-      } catch (e) {
+      } catch {
         // Continue...
```

Comment:
I'm always weary if silence *any* type of error. I too often experience time wasted when some issue goes silenced and we don't know where it is. Funny enough the emoji script was a recent example in this very project.

Perhaps we should at least console.* any error just in case. Or better would be, if we know what error we want to skip, we should specifically skip that error. That way actually-unexpected errors won't accidentally be silenced.

Reply:
I think @trusktr suggestion is reasonable.
It more likely the code style or best practice thing:
- Don't make the Exception silence.
- Let it broken.

Based on this PR's points, I think the change it's fine, cos this is the Lint PR for formatting current codebase (not a purely refine/refactor one).
And we could raise another topic/issue to drive going through the whole project to make sure all those parts have a `console` in `warn` or `error` at least.

Reply:
I agree that this function could be improved by sharing *additional* exception details if/when exceptions occur. 

The existing `waitForFunction` helper behavior already causes execution to halt and an error message to be displayed when the returned promise rejects, but error details are not easily accessible (devs would need to include their own `try/catch` and log errors within the function being passed to `waitForFunction`). It would be better to provide error details if/when the promise rejects. This will make it easier to determine if `waitForFunction` rejected because the test function never returned a truthy value or because the test function caused an unexpected error.

It's good feedback. However, this is not the right time or place to discuss it. The code in question is **four years old** and has nothing to do with the lint configuration updates contained in this PR (as @Koooooo-7  has pointed out).

That said, I've updated the `waitForFunction` to address all concerns so we can (hopefully) move forward.

The `waitForFunction` helper now captures the last error within the `catch` block and makes it available if/when the promise returned by `waitForFunction` rejects. Execution should *not* be halted nor should errors be logged to the console within the `catch` block because 1) halting would defeat the purpose of the helper (in this context errors are expected and details are uninteresting compared to the promise being resolve/rejected) and 2) logging errors resulting from each function invocation would spam the console unnecessarily.

Here's what it looked before the update when the `waitForFunction` was rejected as a result of an exception in the test function:

![CleanShot 2024-05-27 at 00 49 32@2x](https://github.com/docsifyjs/docsify/assets/442527/0f227122-c793-457b-98b6-099727824e5b)

Here's what it looks like now:

![CleanShot 2024-05-27 at 00 51 39@2x](https://github.com/docsifyjs/docsify/assets/442527/d1bfff31-d785-41f4-81df-8f7b82a71715)

@trusktr -- Please review so we can approve, merge, and move on to other tasks. Thx.

Reply:
Issue made to check on that later:

- https://github.com/docsifyjs/docsify/issues/2442